packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -2126,8 +2127,8 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
     // If the platform is Android and the text is read only, try to select the
     // previous word if there is one; otherwise, select the single whitespace at
     // the position.
-    if (TextLayoutMetrics.isWhitespace(plainText.codeUnitAt(effectiveOffset))
-        && effectiveOffset > 0) {
+    if (effectiveOffset > 0
```

Comment:
If there's no clear repro, could you add an assert here so we could debug what happened?

Reply:
I worry that there is a legit case where effectiveOffset can be less than 0, and the correct course of action is just to skip this `if`.  Should I leave it with no assertion so that developers aren't confused if they trigger it in a legitimate case?

Reply:
So the theory is that `_textPainter.getPositionForOffset` is producing a `TextPosition` of `(0, upstream)` ? That's not supposed to be a valid output of the `getPositionForOffset` method I guess. If that's the case this uncovers an upstream bug so I think it's still nice to have an assert here. It won't be an issue in release builds unlike the OOB crash.